lib/json5.js

```diff
@@ -59,14 +71,20 @@ JSON5.parse = (function () {
 // If a c parameter is provided, verify that it matches the current character.
 
             if (c && c !== ch) {
-                error("Expected '" + c + "' instead of '" + ch + "'");
+                error("Expected " + renderChar(c) + " instead of " + renderChar(ch));
             }
 
 // Get the next character. When there are no more characters,
 // return the empty string.
 
             ch = text.charAt(at);
-            at += 1;
+            at++;
+            columnNumber++;
+            var ch2 = text.charAt(at);
+            if (ch === '\n' || ch === '\u2028' || ch === '\u2029' || ch === '\r' && ch2 !== '\n') {
```

Comment:
The rule here is simply taken from the ES5 LineTerminatorSequence production - https://es5.github.io/#x7.3 - we do indeed recognise \r as a valid newline as the test case verified, but we need to ensure not to recognise \r\n as two newlines.